### PR TITLE
demo app: add cross button in the input fields in app settings 

### PR DESCRIPTION
**Pull request by khairul-alam-licon**
# Description
Add cross button to clear texts easily in the input fields of project id and subscription key in demo app settings. 
It will help to copy-paste a new id values quickly. 

note: maybe it can be excluded in next release.

# Screenshot
Screen Shot 2021-07-15 at 16 32 17

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt [hunk 1]
```diff
@@ -132,6 +132,8 @@ class SettingsMenuActivity : BaseActivity() {
             QASettingsActivity.start(this@SettingsMenuActivity)
         }
 
+        binding.clearProjectId.setOnClickListener { binding.editProjectId.setText("") }
+        binding.clearSubscriptionKey.setOnClickListener { binding.editSubscriptionKey.setText("") }
```
Comment (Climbatize) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt:136]: If you want to stick with this solution I suggest this change:

```suggestion
        binding.clearProjectId.setOnClickListener { binding.editProjectId.getText().clear() }
        binding.clearSubscriptionKey.setOnClickListener {  binding.editProjectId.getText().clear() }
```

File: testapp/src/main/res/layout/settings_menu_activity.xml [hunk 2]
```diff
@@ -58,33 +58,61 @@
...
+
+                    <androidx.appcompat.widget.AppCompatImageView
+                        android:id="@+id/clearSubscriptionKey"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_gravity="end"
+                        android:layout_marginTop="@dimen/small_4"
+                        android:layout_marginEnd="@dimen/small_10"
+                        android:src="@drawable/ic_close_24dp" />
+                </FrameLayout>
```
Comment (Climbatize) [inline testapp/src/main/res/layout/settings_menu_activity.xml:115]: I don't if you can use this more actual solution with `endIconMode`:

```xml
<com.google.android.material.textfield.TextInputLayout
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:hint="@string/hint_text"
    app:endIconMode="clear_text">

  <com.google.android.material.textfield.TextInputEditText
      android:layout_width="match_parent"
      android:layout_height="wrap_content"/>

</com.google.android.material.textfield.TextInputLayout>
```

Comment (khairul-alam-licon) [reply testapp/src/main/res/layout/settings_menu_activity.xml:115]: Yeah, I tried this solution before but it's breaking the UI when there is invalid input, so I am doing my own way :D 
Screen Shot 2021-07-13 at 12 28 47

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt (hunk 1 above)
Comment (khairul-alam-licon) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt]: Updated.

File: testapp/src/main/res/layout/settings_menu_activity.xml (hunk 2 above)
Comment (Climbatize) [reply testapp/src/main/res/layout/settings_menu_activity.xml:115]: you'd rather fix the invalid input icon placement in my opinion :)

Comment (khairul-alam-licon) [reply testapp/src/main/res/layout/settings_menu_activity.xml]: Updated this part.

**Review by munir-rakuten — APPROVED** [review]
Looks good to me 👍

File: testapp/src/main/res/layout/settings_menu_activity.xml (hunk 2 above)
Comment (Climbatize) [reply testapp/src/main/res/layout/settings_menu_activity.xml]: can you show a screenshot? The XML looks way better 👍

Comment (khairul-alam-licon) [reply testapp/src/main/res/layout/settings_menu_activity.xml]: @Climbatize I have updated the screenshot in the PR's description. 
Yeah, I like this clean solution, Thank you for your suggestions.

**Review by Climbatize — APPROVED** [review]
Nice job, very clean

**Pull request merged**